src/main/java/homework_nr_12/FuelTypes.java

```diff
@@ -0,0 +1,11 @@
+package homework_nr_12;
+
+public enum FuelTypes {
+    BENZIN(true),
+    SOLARA(true),
+    PROPAN(true),
+    METAN(false);
+    private final boolean isLiquid;
+    FuelTypes(boolean isLiquid){this.isLiquid = isLiquid;}
+    boolean isLiquidFuel(){return this.isLiquid;}
```

Comment:
there is a better way without adding a boolean variable.

hint: `if(? == METAN)`

Reply:
Sure. But only in this case, when only METAN is not liquid. And if you need to add other kinds of fuel this way will be worse. For example, if you add HYDROGEN(false), SYNGAS(false), WOOD(false), COAL(false) and CHARCOAL(false), you will need to write something like if(this == METAN || this == HYDROGEN || this == SYNGAS) and so on. When adding a new fuel type you will also need to edit boolean isLiquidFuel() method. By excluding boolean variable your class loses his flexibility.

Reply:
Yes you're right. In this case it will be better without the variable but in the future if the application grows the variable will be better. Just wanted you to understand how the this keyword will work in an enum or a class. I'll approve that